### ISSUE TITLE
Practical Foundations for Programming Languages  2nd Edition (preview)

### ISSUE THREAD

**Issue by harrykar** [EbookFoundation/free-programming-books#3002]
http://www.cs.cmu.edu/~rwh/pfpl/2nded.pdf

**Comment by eshellman** [comment]
It looks like you understand the contribution guidelines - these suggestions all look good. PR's would be much appreciated.

**Comment by harrykar** [comment]
PR's

**Comment by harrykar** [comment]
I just have forked your repo and cloned it locally now what do?

**Comment by harrykar** [comment]
If someone from staff read here please explain the required steps (what files fill and how) for do a PR as i said in a previous post i just have a copy of your repo(or beter a set of templates i guess) Thanks

**Comment by ShipsWithCannons** [comment]
Please take a look at the HOWTO.md, which can be found [here](https://github.com/EbookFoundation/free-programming-books/blob/master/HOWTO.md).

**Comment by eshellman** [comment]
Don't be afraid to make mistakes, we do our best to welcome new contributors.

**Comment by harrykar** [comment]
thank you both i just start reading if i have questions i refer here

**Comment by Tlalanne** [comment]
Hey sorry to intrude on this conversations but I notice that there wasn’t much on web hacking and penetration testing .

**Comment by harrykar** [comment]
few questions
------------------
a) 

> insert your links in alphabetical order.

What mean exactly? Perhaps  alphabetical order refers to the first letter of the text? 
e.g. 
1. Absolute space
2. Hierarchy tree 

b)
suppose i have to propose a book written in English 
1. i create a branch in my local repo say harrykar 
2.  open the  free-programming-books-en.md file in there and put my proposed book in alphabetical order (respect of the first title letter in list)  
...
[MyBook] - MyAuthor PDF
...
3. i commit to my local repo 
4. i push to my remote forked github repo
5. somehow(dunno actually how but i guess it's not hard ) i create a PR
You can confirm if that process is correct?

**Comment by harrykar** [comment]
@Tlalanne You can open a new issue i guess see up 2nd tab from left

**Comment by eshellman** [comment]
> ## few questions
> a)
> 
> > insert your links in alphabetical order.
> 
> What mean exactly? Perhaps alphabetical order refers to the first letter of the text?
> e.g.

it's the case-insensitive unicode collation of the title string, to be specific. so 1Title < aTitle < Title

> b)
> suppose i have to propose a book written in English
> 
> 1. i create a branch in my local repo say harrykar
> 2. open the  free-programming-books-en.md file in there and put my proposed book in alphabetical order (respect of the first title letter in list)
>    ...
>    [MyBook] - MyAuthor PDF
>    ...
> 3. i commit to my local repo
> 4. i push to my remote forked github repo
> 5. somehow(dunno actually how but i guess it's not hard ) i create a PR
>    You can confirm if that process is correct?

correct

**Comment by harrykar** [comment]
after step 3. in my earlier post i have to merge in master or merging responsibility is yours (i observe the alphabetical order in insert the book in the list)? In other words you want delivered the master updated(that's seems to me more logical) or the original master and my branch?

**Comment by harrykar** [comment]
That one just exists under Theoretical Computer Science section
